scripts/gateway-verification/verify_provider_cleanup_on_failure.py

```
@@ -0,0 +1,157 @@
+            env={"LIGHTSPEED_PROVIDER": "openai", "LIGHTSPEED_MODEL": "gpt-4o-mini"},
+            read_only=False,
+            credential_secret_name=CREDENTIAL_SECRET_NAME,
+        )
+        print(
+            "FAIL (1): spawn() succeeded -- expected it to fail (FAILURE_IMAGE has no "
+            "lightspeed_agentic server, so HTTP readiness should time out)"
+        )
+        sys.exit(1)
+    except RuntimeError as exc:
+        print("PASS (1): spawn() failed as expected:\n   ", exc)
+    except Exception as exc:  # pragma: no cover - diagnostic path
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)